lib/grppccom.gi

```diff
@@ -1207,15 +1207,51 @@ end);
 #M  ComplementcClassesRepresentatives( <G>, <N> )
 ##
 InstallMethod( ComplementClassesRepresentatives,
-  "tell that the normal subgroup must be solvable",IsIdenticalObj,
-  [IsGroup,IsGroup],-2*RankFilter(IsGroup),
+  "tell that the normal subgroup or factor must be solvable", IsIdenticalObj,
+  [ IsGroup, IsGroup ], -2*RankFilter(IsGroup),
 function( G, N )
-  if IsSolvableGroup(N) then
+  if IsSolvableGroup(N) or HasSolvableFactorGroup(G, N) then
     TryNextMethod();
   fi;
-  Error("Cannot compute complement classes for nonsolvable normal subgroups");
+  Error("Cannot compute complements if both N and G/N are nonsolvable");
 end);
 
+
+#############################################################################
+##
+#M  ComplementcClassesRepresentatives( <G>, <N> ) . from conj. cl. subgroups
+##
+InstallMethod( ComplementClassesRepresentatives,
+  "using conjugacy classes of subgroups", IsIdenticalObj,
+  [ IsGroup and HasConjugacyClassesSubgroups, IsGroup ], 0,
+
+function( G, N )
+
+  local C, Hc, H;
+
+  # if <N> is trivial the only complement is <G>
+  if IsTrivial(N) then
+      C := [ G ];
+  # if <G> and <N> are equal the only complement is trivial
+  elif G = N  then
+      C := [ TrivialSubgroup(G) ];
+  elif not IsNormal(G, N) then
+    Error("N must be normal in G");
```

Comment:
This error, and the case G=N above, are not currently covered by the tests, see https://codecov.io/gh/gap-system/gap/pull/563/compare